src/tests/ftest/launch.py

```diff
@@ -128,11 +128,42 @@ def set_test_environment():
     usr_sbin = os.path.join("/usr/", "sbin")
     path = os.environ.get("PATH")
 
+    # Get the default interface to use if OFI_INTERFACE is not set
+    print("Detecting network devices")
+    available_interfaces = {}
+    # Find all the /sys/class/net interfaces on the launch node (excluding lo)
+    net_path = os.path.join(os.path.sep, "sys", "class", "net")
+    for device in sorted([dev for dev in os.listdir(net_path) if dev != "lo"]):
+        # Get the interface state - only include active (up) interfaces
+        with open(os.path.join(net_path, device, "operstate"), "r") as buffer:
+            state = buffer.read().strip()
+        # Get the interface speed - used to select the fastest available
+        with open(os.path.join(net_path, device, "speed"), "r") as buffer:
+            speed = int(buffer.read().strip())
+        print(
+            "  - {0:<5} (speed: {1:>6} state: {2})".format(
+                device, speed, state))
+        # Only include the first active interface for each speed - first is
+        # determined by an alphabetic sort: ib0 will be checked before ib1
+        if state.lower() == "up" and speed not in available_interfaces:
+            available_interfaces[speed] = device
+    try:
+        # Select the fastest active interface available by sorting the speeds
+        interface = available_interfaces[sorted(available_interfaces)[-1]]
+    except IndexError:
+        print(
+            "Error obtaining a default interface from: {}".format(
+                os.listdir(net_path)))
+        exit(1)
+    print(
+        "Using {} as the default interface from {}".format(
+            interface, available_interfaces))
+
     # Update env definitions
     os.environ["PATH"] = ":".join([bin_dir, sbin_dir, usr_sbin, path])
     os.environ["DAOS_SINGLETON_CLI"] = "1"
     os.environ["CRT_CTX_SHARE_ADDR"] = "1"
-    os.environ["OFI_INTERFACE"] = os.environ.get("OFI_INTERFACE", "eth0")
+    os.environ["OFI_INTERFACE"] = os.environ.get("OFI_INTERFACE", interface)
```

Comment:
Doesn't this mean that the interface selected in the above search is only used if `OFI_INTERFACE` is not set in the environment?  If so, then https://github.com/daos-stack/daos/blob/07b1dcf742c93151046e2199b6e60836b84d9596/ftest.sh#L206 might be relevant.

Reply:
So this means that we are actually not using ib0?

Reply:
That's my suspicion, which is why I wanted to ask/point it out.  The 2h15m hardware test run time for this PR suggests the same.  I did a run of this patch, but also setting `OFI_INTERFACE=ib0` for the hardware run over the weekend and it ran for 1h57m instead of the 2h15m of master (and this PR).

Reply:
Yes, it would appear that the export in `ftest.sh` is overriding the default with `eth0` even though launch.py is selecting `ib0`:
```
PARAMS (key=fabric_iface, path=/run/server_config/servers/*, default=eth0) => 'eth0'
```
This behavior - using the externally set OFI_INTERFACE value - is as designed, to allow the user to override the value determined by launch.py.   I just missed the fact that ftest.sh was exporting an OFI_INTERFACE value.  Applying the fix now.